test/e2e/metering_manual_install_tests.go

```diff
@@ -107,6 +112,137 @@ func testManualMeteringInstall(
+	service := &corev1.Service{}
+	storageClass := &v1beta1.StorageClass{}
+
+	for name, file := range files {
+
+		switch name {
+		case "pod":
+			if err := deploy.DecodeYAMLManifestToObject(file, server); err != nil {
+				return err
+			}
+			ctx.Client.CoreV1().Pods("nfs").Create(context.TODO(), server, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		case "service":
+			if err := deploy.DecodeYAMLManifestToObject(file, service); err != nil {
+				return err
+			}
+			ctx.Client.CoreV1().Services("nfs").Create(context.TODO(), service, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		case "storageClass":
+			if err := deploy.DecodeYAMLManifestToObject(file, storageClass); err != nil {
+				return err
+			}
+			ctx.Client.StorageV1beta1().StorageClasses().Create(context.TODO(), storageClass, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		}
+	}
+	// wait and Poll , get service
+	svc, err := ctx.Client.CoreV1().Services("nfs").Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+	if apierrors.IsNotFound(err) && svc.Spec.ClusterIP == ""{
+
+		err = wait.Poll(3*time.Second, 5*time.Minute, func() (bool, error) {
+
+			svc, err = ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+
+			return svc.Spec.ClusterIP != "", err
+
+		})
+
+	}
+
+//decode pvc
+//pvc.Spec.nfs.server = svc.Spec.ClusterIP
+//create ie
+	type nfsPersistentVolume struct {
+				CRD *apiextv1.CustomResourceDefinition
+			}
+
+			var nfsManifestPersistentVolumeDictionary = map[string]nfsManifestPersistentVolume{
+				"nfsPersistentVolume":
+				{
+svc.Spec.ClusterIP = clusterIP
+					resourceType: nfsPersistentVolume{},
+					resourceFile: "/test/testhelpers/nfs_persistentvolume.yaml",
+					actions:      []string{"CREATE"},
+				},
+			}
+
+			//  - Decode the NFS PV YAML manifest
+			//get a go object
+			// after we get clusterIP, save it in a var, pvc.spec.nfs.server = val
+			err = deploy.DecodeYAMLManifestToObject(nfsManifestPersistentVolumeDictionary["nfsPersistentVolume"].resourceFile, nfsManifestPersistentVolumeDictionary["nfsPersistentVolume"].resourceType)
+			if err != nil {
+				return fmt.Errorf("failed to decode the %s YAML service manifest: %v", nfsManifestPersistentVolumeDictionary["nfsPersistentVolume"], err)
+
+				// Update the NFS PV configuration w/ the populated spec.ClusterIP from the NFS Service
+			}
+
+			decodedPersistentVolumeYaml := nfsManifestPersistentVolumeDictionary["nfsPersistentVolume"].resourceType
+
+
+			decodedPersistentVolumeYAML.Spec.NFS.Server := clusterIP
+			_, err = ctx.Client.CoreV1().Namespaces().Create(context.Background(), &nfsPersistentVolume, metav1.CreateOptions{})
+			if err != nil {
+				return err
+			}
+			ctx.Logger.Debugf("Created the %s nfs_clusterIP", clusterIP)
+
+			return err
+
+
+}
```

Comment:
I would shy away from this kind of implementation, and just try and emulate what we worked on offline for how to handle the other resources. Instead, decode the persistent volume into a Go variable, update that variable's `spec.NFS.Server = svc.Spec.ClusterIP` we got previously and pass that now updated object using the following client-go API call:

```go
	err = ctx.Client.CoreV1().PersistentVolumes().Create(context.TODO(), pv, metav1.CreateOptions{})
```

Reply:
Agree. Have simplified.